src/config.c

```diff
@@ -204,34 +209,35 @@ CONFIG_GETTER(getWorkThreads) {
   return sdscatprintf(ss, "%lu", config->numWorkerThreads);
 }
 
-// MT_MODE
-
-CONFIG_SETTER(setMtMode) {
-  const char *mt_mode;
-  int acrc = AC_GetString(ac, &mt_mode, NULL, 0);
+// MIN_OPERATION_WORKERS
+CONFIG_SETTER(setMinOperationWorkers) {
+  size_t newNumThreads;
+  int acrc = AC_GetSize(ac, &newNumThreads, AC_F_GE0);
   CHECK_RETURN_PARSE_ERROR(acrc);
-  if (!strcasecmp(mt_mode, "MT_MODE_ONLY_ON_OPERATIONS")){
-    config->mt_mode = MT_MODE_ONLY_ON_OPERATIONS;
-  } else if (!strcasecmp(mt_mode, "MT_MODE_FULL")){
-    config->mt_mode = MT_MODE_FULL;
-  } else {
-    QueryError_SetError(status, QUERY_EPARSEARGS, "Invalid MT mode");
-    return REDISMODULE_ERR;
+  if (newNumThreads > MAX_WORKER_THREADS) {
+    return errorTooManyThreads(status);
   }
+  config->minOperationWorkers = newNumThreads;
+  // Will only change the number of workers if we are in an event,
+  // and `numWorkerThreads` is less than `minOperationWorkers`.
+  workersThreadPool_SetNumWorkers();
```

Comment:
doesn't `workersThreadPool_OnEventStart` calls `workersThreadPool_SetNumWorkers`?

Reply:
It does, but also a configuration change should trigger this function